DDDEastAnglia/Views/Account/_ExternalLoginMethodsPartial.cshtml

```diff
@@ -0,0 +1,12 @@
+﻿@model IEnumerable<DDDEastAnglia.Models.LoginMethodViewModel>
+
+@foreach (var externalLogin in Model)
+{
+    <li>
+        @using (Html.BeginForm("ExternalLogin", "Account", new { provider = externalLogin.ProviderName, returnUrl = Url.Action("Index", "Home") }, FormMethod.Post, new { id = externalLogin.ProviderName + "Loginform" }))
```

Comment:
Elsewhere the `returnUrl` is set via the `ViewBag`, but here it is hard-coded to the homepage. That doesn't seem right to me - someone might be on the submissions page, choose to log in to submit, and end up on the home page again.